ui/src/app/runtime/runtime-apps/runtime-apps.component.ts

```diff
@@ -0,0 +1,100 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {RuntimeApp} from '../model/runtime-app';
+import {Page} from '../../shared/model/page';
+import {RuntimeAppsService} from '../runtime-apps.service';
+import {BsModalService, BsModalRef} from 'ngx-bootstrap';
+import {Observable} from 'rxjs/Observable';
+import {RuntimeAppComponent} from '../runtime-app/runtime-app.component';
+import {Subscription} from 'rxjs/Subscription';
+
+/**
+ * Component that loads Runtime applications.
+ *
+ * @author Ilayaperumal Gopinathan
+ * @author Vitrac Damien
+ */
+@Component({
+  selector: 'app-runtime-apps',
+  templateUrl: './runtime-apps.component.html',
+})
+export class RuntimeAppsComponent implements OnInit, OnDestroy {
+
+  /**
+   * Observable of a runtime applications page
+   */
+  runtimeApps$: Observable<Page<RuntimeApp>>;
+
+  /**
+   * Subscription of Runtime Apps
+   */
+  subscriptionRuntimeApps: Subscription;
+
+  /**
+   * Pagination state
+   * @type {{page: number; size: number}}
+   */
+  pagination = {page: 0, size: 10};
```

Comment:
See my other comment - I think this is too generic and would like something more typed - I like concrete classes but in this case an Interface should be fine, e.g:

```
export interface Pagination {
  page: number;
  size: number;
} 
const pagination: Pagination = { page:1, size:10 }
```

Then you can at least say: `pagination: Pagination = {page: 0, size: 10}; `

(And if somebody fatfingers something else in between the curly braces, the compiler will at least complain.)

Reply:
Yes. The interface is in the PR 612 (Pagination, Filter, Sort Interface).